flink-python/src/main/java/org/apache/flink/table/runtime/arrow/ArrowUtils.java

```diff
@@ -223,39 +239,43 @@ private static Field toArrowField(RowType.RowField rowField) {
 
 	private static ArrowFieldWriter<BaseRow> createBaseRowArrowFieldWriter(FieldVector vector, LogicalType fieldType) {
```

Comment:
The return type now may not be a BaseRow. Maybe also add a test in `BaseRowArrowReaderWriterTest` to cover this case？

Reply:
Good catch! Updated the result type. However, have not added a new test as this is already covered in the BaseRowArrowReaderWriterTest.